### PR TITLE
chgres_cube - remove wgrib2 from the GRIB2 data read routines.

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## DESCRIPTION OF CHANGES: 

Begin removal of WGRIB2 from chgres_cube:
- Update routine `read_input_atm_grib2_file`, `read_input_sfc_grib2_file`,  `read_winds` and `read_grib_soil` to read input GRIB2 files using G2LIB instead of WGRIB2.
- Update routine `define_input_grid_gfs_grib2` to flip the pole of GFS GRIB2 data. WGRIB2 and G2LIB use different conventions for defining global gaussian grids.
- Update to G2LIB v3.4.3 or higher, which eliminated occasional segmentation faults and slow wall clock times.
- Two new units test for GRIB2 read routines `read_input_atm_grib2_file` and `read_input_sfc_grib2_file`.

## TESTS CONDUCTED: 
Two new unit tests were created. 

All consistency tests were run on all supported machines. The GRIB2 tests failed as expected. But differences between the baseline files and the test files were small. Will provide location of all log files shortly.

## DEPENDENCIES:
None.

## DOCUMENTATION:
N/A

## ISSUE: 
Part of #591. This is too big a project for one PR. Additional PRs under that issue will follow.

### CODE REVIEW

**Comment by kgerheiser** [comment]
This will simplify library maintenance tremendously. Installing and maintaining wgrb2 as a library (and executable) was a real pain.

**Comment by GeorgeGayno-NOAA** [comment]
Logs from the failed consistency tests are on Hera: /scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/wgrib2_logs

I can provide the output files and their differences from the baseline for visualizing if you wish.

**Comment by GeorgeGayno-NOAA** [comment]
> Logs from the failed consistency tests are on Hera: /scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/wgrib2_logs
> 
> I can provide the output files and their differences from the baseline for visualizing if you wish.

I wrote some scripts to create some difference plots. All plots should be done by next Tuesday.

**Comment by GeorgeGayno-NOAA** [comment]
All test output is on Hera - /scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube_nowgrib2

There is a subdirectory for each failed GRIB2 consistency test. The directory contains the output log file from the test, which lists all fields that differ from the baseline and a summary of those differences (via the `nccmp` utility).  Under the ./graphics subdirectory there are some Grads plots of difference fields.

@JeffBeck-NOAA and @LarissaReames-NOAA - please take a look. Let me know if you want to see any other plots. And recall, this PR is only for the conversion of the read_data module and not the model_grid module. So there are no lat/lon difference plots.

File: sorc/chgres_cube.fd/input_data.F90 [hunk 1]
```diff
@@ -4842,665 +5211,925 @@ subroutine read_input_sfc_grib2_file(localpet)
...
    endif
 
+   if (.not. minmax_vgfrc_from_climo) then
+
+     if (localpet == 0) then
+
+       print*,"- READ MIN VEG FRACTION."
+
+       jdisc   = 2  ! Search for discipline - land products
+       j = 1105 ! grib2 file does not distinguish between the various veg
```
Comment (LarissaReames) [inline sorc/chgres_cube.fd/input_data.F90:5660]: I don't know if we want to address this in the current PR, or open a new one after this one is approved, but in the more recent HRRR files they actually have upgraded to using the correct wgrib2 parameter numbers for min and max veg fraction (231 and 232, respectively). Do we want to put that in as the first try for these variables here or wait and put it in a new PR? It wouldn't change any of the baseline results as the current HRRR file being used for baseline testing is old so it still needs these record number specifications. I'm honestly not sure how we haven't had an Issue submitted due to failure caused by the upgrade to the newer (and better) parameter numbers.

Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/input_data.F90:5660]: I might do this in a separate PR. 

Ideally, I would like all these static fields created at the grid generation step (via sfc_climo_gen). Do we have the original greenness dataset used by HRRR? Converting it to the format used by sfc_climo_gen is not difficult.

**Review by JeffBeck-NOAA — APPROVED** [review]
Difference fields between wgrib2 and g2 results look reasonable (only noise) and on par with what is expected for nearly identical input grid point locations.

**Comment by GeorgeGayno-NOAA** [comment]
Thanks, @JeffBeck-NOAA. I will merge now.

**Pull request merged**